core/spec/models/spree/image/preferences_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'spree/testing_support/preferences'
+
+# Use a string to avoid autoloading Spree::Image, which would load the
+# default config values.
+RSpec.describe 'Spree::Image', type: :model do
+  custom_styles = {
+    mini: '48x48>',
+    small: '100x100>',
+    product: '240x240>',
+    large: '600x600>',
+    jumbo: '1200x1200>'
+  }
+  custom_default_style = :other
+
+  before {
+    expect(Spree::Config.product_image_styles).not_to eq(custom_styles)
+    expect(Spree::Config.product_image_style_default).not_to eq(custom_default_style)
+
+    # We have to set the `product_image_styles` before we autoload
+    # Spree::Image (which would load the default values for
+    # `product_image_styles` and `product_image_style_default`).
+    stub_spree_preferences(
+      product_image_styles: custom_styles,
+      product_image_style_default: custom_default_style
+    )
+  }
+
+  context 'When using Spree::Image::ActiveStorageAttachment as Spree::Config.image_attachment_module' do
+    before { stub_spree_preferences(image_attachment_module: Spree::Image::ActiveStorageAttachment) }
+
+    it 'correctly sets the image styles' do
+      expect(Spree::Image.attachment_definitions[:attachment][:styles]).to eq(custom_styles)
+      expect(Spree::Image.attachment_definitions[:attachment][:default_style]).to eq(custom_default_style)
+    end
+  end
+
+  context 'When using Spree::Image::PaperclipAttachment as Spree::Config.image_attachment_module' do
+    before { stub_spree_preferences(image_attachment_module: Spree::Image::PaperclipAttachment) }
```

Comment:
Similarly as to how the changes to `product_image_styles` are not reflected (because `Spree::Image` has already been loaded, the change in `image_attachment_module` is also ignored.  Therefore this test context runs through the same steps as the one for `Spree::Image::ActiveStorageAttachment`